packages/xstate-wallet/src/integration-tests/basic.test.ts

```diff
@@ -0,0 +1,170 @@
+              amount: '0x06f05b59d3b20000'
+            }
+          ]
+        }
+      ],
+      appDefinition: '0x430869383d611bBB1ce7Ca207024E7901bC26b40',
+      appData: '0x0' // TODO: This works for now but will break when we start validating
+    }
+  };
+}
+
+it('works', async () => {
+  const playerA = new Player('0x275a2e2cd9314f53b42246694034a80119963097e3adf495fbf6d821dc8b6c8e');
+  const playerB = new Player('0x3341c348ea8ade1ba7c3b6f071bfe9635c544b7fb5501797eaa2f673169a7d0d');
+
+  playerA.channelWallet.onSendMessage(message => {
+    if (isNotification(message) && message.method === 'MessageQueued') {
+      const pushMessageRequest = generatePushMessage(
+        message.params,
+        playerB.participantId,
+        playerA.participantId
+      );
+      playerB.channelWallet.pushMessage(pushMessageRequest);
+    }
+  });
+  playerB.channelWallet.onSendMessage(message => {
+    if (isNotification(message) && message.method === 'MessageQueued') {
+      const pushMessageRequest = generatePushMessage(
+        message.params,
+        playerA.participantId,
+        playerB.participantId
+      );
+      playerA.channelWallet.pushMessage(pushMessageRequest);
+    }
+  });
+
+  const createEvent = generateCreateChannelRequest(playerA.participant, playerB.participant);
+
+  const playerAResponsePromise = createPromise(
+    playerA.messagingService.outboxFeed.pipe(
+      filter(m => 'id' in m && m.id === createEvent.id),
+      map(m => m as CreateChannelResponse)
+    )
+  );
+
+  await playerA.messagingService.receiveMessage(createEvent);
+
+  playerA.channelWallet.workflows[0].machine.send({type: 'USER_APPROVES'});
+
+  const response = await playerAResponsePromise;
+  expect(response.result).toBeDefined();
+  const {channelId} = response.result;
+
+  const joinEvent: JoinChannelRequest = generateJoinChannelRequest(channelId);
+
+  const playerBResponsePromise = createPromise(
+    playerB.messagingService.outboxFeed.pipe(
+      filter(r => 'id' in r && r.id === joinEvent.id),
+      map(r => r as JoinChannelResponse)
+    )
+  );
+
+  await playerB.messagingService.receiveMessage(joinEvent);
+
+  // Wait for the create channel service to start
+  await waitForExpect(async () => {
+    expect(playerB.workflow?.machine.state.value).toMatchObject({
+      confirmJoinChannelWorkflow: 'invokeCreateChannelConfirmation'
+    });
+  }, 3000);
+
+  playerB.channelWallet.workflows[0].machine.send({type: 'USER_APPROVES'});
+
+  const playerBResponse: JoinChannelResponse = await playerBResponsePromise;
+  expect(playerBResponse.result).toBeDefined();
+});
+
+class Player {
+  privateKey: string;
+  get signingAddress() {
+    return new Wallet(this.privateKey).address;
+  }
+  store: Store;
+  messagingService: MessagingServiceInterface;
+  channelWallet: ChannelWallet;
+
+  get workflow(): Workflow | undefined {
+    return this.channelWallet.workflows.length > 0 ? this.channelWallet.workflows[0] : undefined;
+  }
```

Comment:
This getter is only used in one place and everywhere else is `channelWallet.workflows[0]`. I don't think this is necessary, because you have use optional chaining to deal with the undefined value.

Reply:
I've changed this to `WorkflowState` to make the tests cleaner.